api-extractor/src/Extractor.ts

```diff
@@ -95,8 +107,35 @@ export default class Extractor {
       throw new Error('Unable to load file: ' + options.entryPointFile);
     }
 
+    // Assign _packageFolder by probing upwards from entryPointFile until we find a package.json
+    const currentPath: string = path.resolve(options.entryPointFile);
+    this._packageFolder = PackageJsonHelpers.findPackagePathUpwards(currentPath);
+
     this.package = new ApiPackage(this, rootFile); // construct members
     this.package.completeInitialization(); // creates ApiDocumentation
+    this.package.visitTypeReferencesForApiItem();
+  }
+
+  /**
+   * Determines if a sourceFile was exported locally. If false, then the sourceFile
+   * was exported from an external package, in relation to the package currently being
+   * analyzing by the Extractor class.
+   */
+  public isLocalSourceFile(sourceFile: ts.SourceFile): boolean {
```

Comment:
If I remember right from our last conversation, the isLocalSourceFile() check is an artifact from the earlier approach.  The new algorithm implicitly knows whether the source file is local or not by walking upwards to find the nearest package.json.  This is all we need to know whether the source file belongs to our current project or not.